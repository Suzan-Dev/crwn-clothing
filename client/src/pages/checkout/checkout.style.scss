.checkout-page {
  width: 95%;
  min-height: 90vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 50px auto 0;

  @media screen and (min-width: 800px) {
    width: 55%;
  }

  .checkout-header {
    width: 100%;
    padding: 10px 0;
    display: flex;
    justify-content: space-between;
    border-bottom: 1px solid darkgrey;

    .header-block {
      text-transform: capitalize;
      width: 25%;

      &:nth-last-child(2) {
        width: 18%;
      }
      &:last-child {
        width: 13%;
      }

      @media screen and (min-width: 800px) {
        width: 23%;

        &:nth-last-child(2) {
          width: 23%;
        }

        &:last-child {
          width: 8%;
        }
      }
    }
  }

  .total {
    margin-top: 30px;
    margin-left: auto;
    font-size: 36px;
  }

  .test-warning {
    color: red;
    font-size: 25px;
    margin: 45px 0;
  }

  button {
    margin-left: auto;
    outline: none;
    margin-bottom: 20px;
  }
}
