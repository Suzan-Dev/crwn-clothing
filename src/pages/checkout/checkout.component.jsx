import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import './checkout.style.scss';

import CheckoutItem from '../../components/checkout-item/checkout-item.component';

import {
  selectCartItem,
  selectCartTotal,
} from '../../redux/cart/cart.selector';

const CheckoutPage = ({ cartItems, cartTotal }) => (
  <div className='checkout-page'>
    <div className='checkout-header'>
      <div className='header-block'>
        <span>Product</span>
      </div>
      <div className='header-block'>
        <span>Description</span>
      </div>
      <div className='header-block'>
        <span>Quantity</span>
      </div>
      <div className='header-block'>
        <span>Price</span>
      </div>
      <div className='header-block'>
        <span>Remove</span>
      </div>
    </div>

    {cartItems.map((cartItem) => (
      <CheckoutItem key={cartItem.id} cartItem={cartItem} />
    ))}

    <div className='total'>
      <span>Total : ${cartTotal}</span>
    </div>
  </div>
);

const mapStateToProps = createStructuredSelector({
  cartItems: selectCartItem,
  cartTotal: selectCartTotal,
});

export default connect(mapStateToProps)(CheckoutPage);
