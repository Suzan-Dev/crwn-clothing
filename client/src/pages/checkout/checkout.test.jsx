import React from 'react';
import { shallow } from '../../setupTests';
import { CheckoutPage } from './checkout.component';

describe('Testing: CheckoutPage component!', () => {
  const mockCartItems = [{ id: 1 }, { id: 2 }];
  const mockCartTotal = 100;

  const mockProps = {
    cartItems: mockCartItems,
    cartTotal: mockCartTotal,
  };

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<CheckoutPage {...mockProps} />);
  });

  test('should render CheckoutPage component.', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  test('should have equal num of CheckoutItem component as objects in cartItems.', () => {
    expect(wrapper.find('Connect(CheckoutItem)').length).toBe(mockCartItems.length);
  });

  test('should pass cartTotal prop to StripeButton component.', () => {
    expect(wrapper.find('StripeButton').prop('price')).toBe(100);
  });
});
