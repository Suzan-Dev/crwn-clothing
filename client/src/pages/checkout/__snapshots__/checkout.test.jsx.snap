// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing: CheckoutPage component! should render CheckoutPage component. 1`] = `
"<div className=\\"checkout-page\\">
  <div className=\\"checkout-header\\">
    <div className=\\"header-block\\">
      <span>
        Product
      </span>
    </div>
    <div className=\\"header-block\\">
      <span>
        Description
      </span>
    </div>
    <div className=\\"header-block\\">
      <span>
        Quantity
      </span>
    </div>
    <div className=\\"header-block\\">
      <span>
        Price
      </span>
    </div>
    <div className=\\"header-block\\">
      <span>
        Remove
      </span>
    </div>
  </div>
  <Connect(CheckoutItem) cartItem={{...}} />
  <Connect(CheckoutItem) cartItem={{...}} />
  <div className=\\"total\\">
    <span>
      Total : $
      100
    </span>
  </div>
  <div className=\\"test-warning\\">
    *Please use the following test credit card for payment*
    <br />
    4242 4242 4242 4242 - Exp: 02/22 - CVV: 123
  </div>
  <StripeButton price={100} />
</div>"
`;
